bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/DescriptorRpc.scala

```diff
@@ -0,0 +1,32 @@
+package org.bitcoins.rpc.client.common
+
+import org.bitcoins.rpc.jsonmodels.{
+  DeriveAddressesResult,
+  GetDescriptorInfoResult
+}
+import play.api.libs.json.{JsString, Json}
+import org.bitcoins.rpc.serializers.JsonSerializers._
+
+import scala.concurrent.Future
+
+/**
+  * RPC calls in V18 that use descriptor to give us output information
+  * @see [[https://bitcoincore.org/en/doc/0.18.0/rpc/util/deriveaddresses/]]
+  * @see [[https://bitcoincore.org/en/doc/0.18.0/rpc/util/getdescriptorinfo/]]
+  */
+trait DescriptorRpc {
+  self: Client =>
+
+  def deriveAddresses(
+      descriptor: String,
+      range: Option[Vector[Double]]): Future[DeriveAddressesResult] = {
+    bitcoindCall[DeriveAddressesResult](
+      "deriveaddresses",
+      List(JsString(descriptor), Json.toJson(range)))
```

Comment:
In the case where `range == None` shouldn't this second param be omitted?

Reply:
Does `Json.toJson` do that automatically?

btw this is just copy paste from the original `v18DescriptorRpc.scala`

Reply:
`Json.toJson(None)` returns `JsNull`. Not sure if this is the intended functionality but I have a feeling that this param list should be a val defined on matching on `range`